mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -0,0 +1,342 @@
+    ds = ds.sel(Time=slice(start_date, end_date))
+
+    return ds  # }}}
+
+
+def _preprocess(ds, calendar, time_variable_name, variable_list, sel_values,
+                isel_values, variable_map, start_date, end_date,
+                year_offset):  # {{{
+    """
+    Overloads the _preprocess function from MpasXarray to include
+    mapping variable names.  See MpasXaray._preprocess for more
+    details.
+
+    Parameters
+    ----------
+    ds : xarray.DataSet object
+        The data set containing an MPAS time variable to be used to build
+        an xarray time coordinate and with variable names to be
+        substituted.
+
+    calendar : {'gregorian', 'gregorian_noleap'}, optional
+        The name of one of the calendars supported by MPAS cores
+
+    time_variable_name : string
+        The name of the time variable (typically 'Time' if using a variable_map
+        or 'xtime' if not using a variable_map)
+
+    variable_list : list of strings
+        If present, a list of variables to be included in the data set
+
+    sel_values : dict
+        A dictionary of coordinate names (keys) and values or arrays of
+        values used to slice the variales in the data set.  See
+        xarray.DataSet.sel() for details on how this dictonary is used.
+        An example:
+            selectCorrdValues = {'cellLon': 180.0}
+
+    isel_values : dict
+        A dictionary of coordinate names (keys) and indices, slices or
+        arrays of indices used to slice the variales in the data set.  See
+        xarray.DataSet.isel() for details on how this dictonary is used.
+        An example:
+            isel_values = {'nVertLevels': slice(0, 3),
+                           'nCells': cellIDs}
+
+    variable_map : dict
+        A dictionary with keys that are variable names used by
+        MPAS-Analysis and values that are lists of possible names for the same
+        variable in the MPAS dycore that produced the data set (which may
+        differ between versions).
+
+    start_date, end_date : string or datetime.datetime
+        If present, the first and last dates to be used in the data set.  The
+        time variable is sliced to only include dates within this range.
+
+    year_offset : float
+        An offset used to convert an MPAS date to a date in the range supported
+        by xarray (numpy.datetime64).  Resulting dates must be between 1678
+        and 2622.
+
+    Returns
+    -------
+    ds : xarray.DataSet object
+        A copy of the data set with the time coordinate set and which
+        has been sliced.
+
+    Authors
+    -------
+    Xylar Asay-Davis
+
+    Last modified
+    -------------
+    02/15/2017
+    """
+
+    submap = variable_map
+
+    # time_variable_names is a special case so we take it out of the map
+    # and handle it manually (adding a new variable rather than renaming
+    # an existing one)
+    if variable_map is not None and \
+            time_variable_name in variable_map:
```

Comment:
I don't know if you need to break this line, it may be clearer (even if the line is long) to keep on one line.